### DISCUSSION TITLE
[API] Too Many Attempts.

### DISCUSSION THREAD

**Discussion by b-keller** [dbarzin/mercator#881] [Q&A]
Bonjour,

J'essaie d'utiliser l'api de Mercator pour synchroniser des éléments depuis un autre application.

après avoir synchronisé avec succès les sites, les bâtiments, les clusters, l'api répond assez rapidement : 
```
{'message': 'Too Many Attempts.'}
```
ce message apparait après moins d'une minute pour environ 62 requêtes

Est-ce qu'il y a un limiteur activé par défaut ? Cette valeur est-elle modifiable ?

Merci
Cordialement

**Comment by dbarzin** [comment]
Il y a une protection DDOS dans Laravel. Tu peux ajouter un sleep dans ton code ou le désactivant dans app/Http/Kernel.php :

```
protected $middlewareGroups = [
    ...
    'api' => [
        'throttle:60,1',
        ...
    ],
];

```

Tu peux commenter ou augmenter la valeur.

**Comment by b-keller** [comment]
En désactivant ce paramètre, ça fonctionne.

ce réglage sera à faire après chaque mise à jour de Mercator ?
Un axe d'amélioration pourrait être de proposer un réglage accessible via l'interface web dans "configuration"  pour la partie api ?

**Comment by dbarzin** [comment]
Oui, si tu changes ce paramètre, il y aura un conflit lors de la mise à jour.
Par contre, si tu ajoutes un `sleep` de 100ms entre chaque accès à l'API tu n'auras plus de problèmes.

**Comment by b-keller** [comment]
en python, ça fonctionne avec un sleep de 500ms
```
import time
time.sleep(500 / 1000)
```
